### DISCUSSION TITLE
Publish jar onto mvn central

### DISCUSSION THREAD

**Discussion by StevenCurran** [kdbinsidebrains/plugin#80] [Ideas]
Hey there. Thanks for the plugin it is awesome! 

I was writing a kerberos wrapper using the useful guide you mentioned for custom auth. It works great, but I had to copy your plugin jar manually into my auth jar repo I created as I could not find your jar for the plugin (and so cred provider / editor classss) via a regular gradle resolver. 

Any chance you could push some of these as releases? 

Thanks again for your hard work!

**Comment by smklimenko** [comment]
I can think about that but see no reason for now. You are the second who's implemented custom auth but for Maven release I have to split the project that will increase the complexity of the project. From other side, auth was changed only 2 times for past 2 years with 100% backward combability so you have to build your jar only once and forever, I'd say so it shouldn't be too hard to get the plugin jar from IDEA and use it locally.

**Reply by StevenCurran** [reply]
So I don't mean to split the credentials code out into it's own jar, I just meant to the plugin binary would be uploaded to mvn central also, instead of just the jetbrains market place. 

Just so we could pull it down from the build. 

But yes in the mean time copying jar locally works.

**Reply by flcong** [reply]
Actually, I came across the same issue..

**Comment by smklimenko** [comment]
ok, if we have already to. Officially it's published to https://s01.oss.sonatype.org/content/repositories/releases/dev/kdbinsidebrains/plugin/ repository but must be synched with Maven Central as well. At least I have updated credentials sample with new dependency and it works fine for me: https://github.com/kdbinsidebrains/credentials/blob/main/pom.xml

        <dependency>
            <groupId>dev.kdbinsidebrains</groupId>
            <artifactId>plugin</artifactId>
            <version>5.0.0</version>
        </dependency>

**Reply by StevenCurran** [reply]
Hmm okay, must just not have synched with our internal artifactory repo. 

Any reason why it cannot be found here? 

https://mvnrepository.com/search?q=dev.kdbinsidebrains 

It is working fine for me with gradle on my home machine, so will need to see why it cannot be accessed in the controlled AF.

**Reply by smklimenko** [reply]
it's there, at least already. Probably you have to check your company policy as you could have only pre/approved artifacts. 
Or, as you didn't see it in UI, could be your company sync it with some XMLs from the same site and you have to wait a few days to get it.
In any case, I'm afraid I can't help you with that.
